src/main/java/org/radarcns/management/repository/filters/UserFilter.java

```diff
@@ -29,23 +29,27 @@ public Predicate toPredicate(Root<User> root, CriteriaQuery<?> query, CriteriaBu
         if (StringUtils.isNotBlank(login)) {
             predicates
                 .add(cb.like(cb.lower(root.get("login")), "%" + login.trim().toLowerCase() + "%"));
+            query.distinct(true);
         }
         if (StringUtils.isNotBlank(email)) {
             predicates
                 .add(cb.like(cb.lower(root.get("email")), "%" + email.trim().toLowerCase() + "%"));
+            query.distinct(true);
         }
 
         if (StringUtils.isNotBlank(projectName)) {
             Join<User, Role> roleJoin = root.join("roles");
             Join<Role, Project> projectJoin = roleJoin.join("project");
             predicates.add(cb.like(cb.lower(projectJoin.get("projectName")),
                     "%" + projectName.trim().toLowerCase() + "%"));
+            query.distinct(true);
         }
         if (StringUtils.isNotBlank(authority)) {
             Join<User, Role> roleJoin = root.join("roles");
             Join<Role, Authority> authorityJoin = roleJoin.join("authority");
             predicates.add(cb.like(cb.lower(authorityJoin.get("name")),
                     "%" + authority.trim().toLowerCase() + "%"));
+            query.distinct(true);
         }
         return predicates.isEmpty() ? null
```

Comment:
Instead of the four statements, perhaps make this an if-else?
```java
if (predicates.isEmpty()) {
    return null
} else {
    query.distinct(true);
    return cb.and(predicates.toArray(new Predicate[0]));
}
```

Reply:
Makes sense. Thanks :)